### PR TITLE
don't run CanZipNamedPipe test on iOS/tvOS

### LINKED ISSUE #62341: [iOS/tvOS] Test Failure System.IO.Compression.Tests.ZipFile_Unix.CanZipNamedPipe
Failure started appearing on the rolling build after https://github.com/dotnet/runtime/pull/62240 landed

```
Assert.Equal() Failure
Expected: 0
Actual:   -1

at System.IO.Compression.Tests.ZipFile_Unix.CanZipNamedPipe()
--- End of stack trace from previous location ---
```

### PR DESCRIPTION
**Pull request by adamsitnik**
fixes #62341

### CODE REVIEW

File: src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs [hunk 1]
```diff
@@ -155,7 +155,7 @@ public async Task CanZipNamedPipe()
             await Task.WhenAll(
                 Task.Run(() =>
                 {
-                    using FileStream fs = new (fifoPath, FileMode.Open, FileAccess.Write, FileShare.Read);
+                    using FileStream fs = new (fifoPath, FileMode.Open, FileAccess.Write, FileShare.Read, bufferSize: 0);
```
Comment (stephentoub) [inline src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs:158]: This is unrelated to iOS and instead for my comments on the previous PR? Just confirming.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs:158]: > This is unrelated to iOS and instead for my comments on the previous PR? Just confirming.

Yes, exactly. I did not want to send a new PR for this small change so I've used the test failure as an opportunity to address the old feedback as well.
